e2e/test/iothub/command/CommandE2ETests.cs

```diff
@@ -0,0 +1,238 @@
+        }
+
+        public static async Task DigitalTwinsSendCommandAndVerifyResponseAsync(string deviceId, string componentName, string methodName, string respJson, string reqJson, MsTestLogger logger)
+        {
+            DigitalTwinClient digitalTwinClient = null;
+            digitalTwinClient = DigitalTwinClient.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+
+            logger.Trace($"{nameof(DigitalTwinsSendCommandAndVerifyResponseAsync)}: Invoke method {methodName}.");
+            Rest.HttpOperationResponse<DigitalTwinCommandResponse, DigitalTwinInvokeCommandHeaders> response = null;
+            if (string.IsNullOrEmpty(componentName))
+            {
+                response =
+                await digitalTwinClient.InvokeCommandAsync(
+                    deviceId,
+                    methodName,
+                    reqJson).ConfigureAwait(false);
+            }
+            else
+            {
+                response =
+                await digitalTwinClient.InvokeComponentCommandAsync(
+                    deviceId,
+                    componentName,
+                    methodName,
+                    reqJson).ConfigureAwait(false);
+            }
+
+
+            logger.Trace($"{nameof(DigitalTwinsSendCommandAndVerifyResponseAsync)}: Method status: {response.Response.StatusCode}.");
+            Assert.AreEqual(200, Shared.StatusCodes.OK, $"The expected response status should be 200 but was {response.Response.StatusCode}");
+            string payload = response.Body.Payload;
+            Assert.AreEqual(respJson, payload, $"The expected response payload should be {respJson} but was {payload}");
+
+            digitalTwinClient.Dispose();
+        }
+
+        public static async Task<Task> SubscribeAndUnsubscribeCommandAsync(DeviceClient deviceClient, string componentName, string methodName, MsTestLogger logger)
+        {
+            var methodCallReceived = new TaskCompletionSource<bool>();
+
+            await deviceClient
+                .SubscribeToCommandsAsync(
+                (request, context) =>
+                {
+                    if (string.IsNullOrEmpty(componentName))
+                    {
+                        logger.Trace($"{nameof(SubscribeAndUnsubscribeCommandAsync)}: DeviceClient method: {request.CommandName}.");
+                    }
+                    else
+                    {
+                        logger.Trace($"{nameof(SubscribeAndUnsubscribeCommandAsync)}: DeviceClient method: {request.ComponentName} {request.CommandName}.");
+                    }
+                    return Task.FromResult(new CommandResponse(new DeviceCommandResponse(), Shared.StatusCodes.OK));
+                },
+                null)
+                .ConfigureAwait(false);
+
+            await deviceClient.SubscribeToCommandsAsync(null, null).ConfigureAwait(false);
```

Comment:
are we simply testing that we can call `SubscribeToCommandsAsync` multiple times/ call it wilh null callback? This seems like something that should be unit tested instead, right?

Reply:
Or if we want to test the whole subscribe -> unsubscribe flow, then we'd have to test that we can subscribe -> receive the callback -> unsubscribe -> should not receive the callback.

Reply:
Actually, yeah, this was some odd refactoring thing. I removed this test as it makes no sense.